csrc/device_lower/lower2device.cpp

```diff
@@ -488,26 +488,26 @@ void GpuLower::lower(Fusion* fusion) {
   dumpExprsIfEnabled(
       exprs_conditional_loops, "generateConditionalFromPredicate");
 
-  const auto exprs_common_index_allocated =
-      allocateCommonScalars(exprs_conditional_loops);
-  dumpExprsIfEnabled(exprs_common_index_allocated, "allocateCommonScalars");
-
```

Comment:
Just moving down. Unrelated to this PR. In general, we should make `allocateCommonScalars` the last pass so that all other passes can benefit from index hoisting.